crates/revm/src/db/states/bundle_state.rs

```diff
@@ -142,9 +148,13 @@ impl BundleState {
         self.contracts.get(hash).cloned()
     }
 
+    // mattsse: what does `substate` mean here exactly?
+
     /// Consume `TransitionState` by applying the changes and creating the reverts
     ///
     /// `with_reverts` flag indicates whether we should collect the reverts.
+
+    // mattsse: what does collecting the reverts mean? keeping the inverse of the transition changes, so they can be reverted/rolled back?
```

Comment:
Yeah in the essence revert is used to rollback the plain state and create the changesets in reth (previous values of change at least).